tests/device_tests/test_session_id_and_passphrase.py

```diff
@@ -54,24 +67,82 @@ def test_session_with_passphrase(client):
     new_session_id = _init_session(client)
     assert new_session_id != session_id
-    assert _get_xpub(client, passphrase="A") == XPUB_PASSPHRASE_A
+    assert _get_xpub(client, passphrase="A") == XPUB_PASSPHRASES["A"]
 
     # Unknown session id is the same as setting it to None.
     _init_session(client, session_id=b"X" * 32)
-    assert _get_xpub(client, passphrase="A") == XPUB_PASSPHRASE_A
+    assert _get_xpub(client, passphrase="A") == XPUB_PASSPHRASES["A"]
+
+
+@pytest.mark.skip_ui
+@pytest.mark.setup_client(passphrase=True)
+def test_multiple_sessions(client):
+    session_ids = []
+
+    # start a session
+    session_ids.append(_init_session(client))
+    assert _get_xpub(client, passphrase="A") == XPUB_PASSPHRASES["A"]
+    # start it again wit the same session id
+    new_session_id = _init_session(client, session_id=session_ids[0])
+    # session is the same
+    assert new_session_id == session_ids[0]
+    # passphrase is not prompted
+    assert _get_xpub(client, passphrase=None) == XPUB_PASSPHRASES["A"]
+
+    # start a second session
+    session_ids.append(_init_session(client))
+    # new session -> new session id and passphrase prompt
+    assert _get_xpub(client, passphrase="B") == XPUB_PASSPHRASES["B"]
+
+    # provide the same session id -> must not ask for passphrase again.
+    new_session_id = _init_session(client, session_id=session_ids[1])
+    assert new_session_id == session_ids[1]
+    assert _get_xpub(client, passphrase=None) == XPUB_PASSPHRASES["B"]
+
+    # provide the first session id -> must not ask for passphrase again and return the same result.
+    new_session_id = _init_session(client, session_id=session_ids[0])
+    assert new_session_id == session_ids[0]
+    assert _get_xpub(client, passphrase=None) == XPUB_PASSPHRASES["A"]
+
+    passphrases = list(XPUB_PASSPHRASES.keys())
+    xpubs = list(XPUB_PASSPHRASES.values())
+    # start as many sessions as the limit is (2 were started already)
+    for i in range(2, SESSIONS_STORED):
+        new_session_id = _init_session(client)
+        assert new_session_id not in session_ids
+        session_ids.append(new_session_id)
+        assert _get_xpub(client, passphrase=passphrases[i]) == xpubs[i]
+
+    # passphrase is not prompted for the started the sessions, regardless the order
+    for i in reversed(range(0, SESSIONS_STORED)):
+        _init_session(client, session_id=session_ids[i])
+        assert _get_xpub(client, passphrase=None) == xpubs[i]
+
+    # creating one more will exceed the limit, the LRU item is at the moment the last one (see above)
+    # it should have been removed -> must ask for passphrase
+    _init_session(client)
+    _get_xpub(client, passphrase="XX")  # create one more
+    _init_session(client, session_id=session_ids[SESSIONS_STORED - 1])
+    _get_xpub(client, passphrase="whatever")  # the session is gone
+
+    # now the second to last is the next LRU
+    _init_session(client)
+    _get_xpub(client, passphrase="XXXX")
```

Comment:
i'm not sure about how exactly to formulate the testcase, but something like ... "create a session that evicts, and then set the same passphrase as with the just-evicted, and check that the eviction did happen"

implementation-wise it is as simple as setting `passphrase="B"` here, but a comment explaining what i'm trying to achieve would help :)

Reply:
I am not sure I understand. We can of course add `passphrase="B"` here instead of `"XXXX"` but what does it test? It tests we prompt for passphrase "B" although it is just being removed. But since we do not send the sessions id and we cache based the sessons id, does it make any sense?

Reply:
Well anyway, it definitely _doesn't hurt_, so I'll add it :).

Reply:
What it seems to me this test set obscures is the fact that _sessions are independent from passphrases_. Using `passphrase="B"` here hints at that, but you are right that it's not really _testing_ anything meaningful.

Perhaps a better choice is something like, create several sessions and not set a passphrase in them?
Or maybe something like this?
```
sessions = []
for _ in range(SESSIONS):
    sessions.append(_init_session(client))
    _get_xpub(client, passphrase="A")
for session_id in sessions:
	_init_session(client, session_id)
	_get_xpub(client, passphrase="A")
```
Do we have a different way to check that there's a different session cache, than using a passphrase?